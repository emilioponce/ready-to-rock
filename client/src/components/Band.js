import React from 'react'
import PropTypes from 'prop-types'
import styles from './Band.module.css'

const Band = ({ title, body, members }) => (
  <div className={styles.band}>
    <h1>{title}</h1>
    <div className={styles.body} dangerouslySetInnerHTML={{ __html: body }} />
    <div>
      <h2>Members:</h2>
      <span className={styles.members}>
      <ul>
        {members
          ? members.map(member => 
                <li key={member.name} className={styles.member}>
                  {member.name} - {member.instrument}
                </li>
              )
          : 'Unknown members'}
          </ul>
      </span>
    </div>
  </div>
)

Band.propTypes = {
  title: PropTypes.string,
  body: PropTypes.string,
  members: PropTypes.array
}

export default Band
