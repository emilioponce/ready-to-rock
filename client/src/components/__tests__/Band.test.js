import React from 'react'
import ReactDOM from 'react-dom'
import Band from '../Band'
import { shallow } from 'enzyme'

const component = (
  <Band
    title="Band name"
    body="<div>bla bla bla</div>"
    members={
      [
        {name:'member1', instrument: 'instrument1'}, 
        {name:'member2', instrument: 'instrument2'}
      ]}
  />
)
describe('Component Band', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('snapshot renders correctly', () => {
    const wrapper = shallow(component)
    expect(wrapper).toMatchSnapshot()
  })
})
