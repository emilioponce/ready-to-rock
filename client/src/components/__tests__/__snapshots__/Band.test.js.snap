// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Band snapshot renders correctly 1`] = `
<div
  className="band"
>
  <h1>
    Band name
  </h1>
  <div
    className="body"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div>bla bla bla</div>",
      }
    }
  />
  <div>
    <h2>
      Members:
    </h2>
    <span
      className="members"
    >
      <ul>
        <li
          className="member"
          key="member1"
        >
          member1
           - 
          instrument1
        </li>
        <li
          className="member"
          key="member2"
        >
          member2
           - 
          instrument2
        </li>
      </ul>
    </span>
  </div>
</div>
`;
